pkg/store/tree.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2019-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/onosproject/onos-config/pkg/store/change"
+	"strconv"
+	"strings"
+)
+
+// BuildTree is a function that takes an ordered array of ConfigValues and
+// produces a structured formatted JSON tree
+func BuildTree(values []*change.ConfigValue) ([]byte, error) {
+
+	root := make(map[string]interface{})
+	rootif := interface{}(root)
+	for _, cv := range values {
+		err := addPathToTree(cv.Path, cv.Value, &rootif)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	buf, err := json.Marshal(root)
+	if err != nil {
+		return nil, err
+	}
+
+	return buf, nil
+}
+
+// addPathToTree is a recursive function that builds up a map
+// suitable for using with json.Marshal, which in turn can be used to feed in to
+// ygot.Unmarshall
+// This follows the approach in https://blog.golang.org/json-and-go "Generic JSON with interface{}"
+func addPathToTree(path, value string, nodeif *interface{}) error {
+	pathelems := strings.Split(path, "/")
```

Comment:
we ought to make "/" a constant in order to be able to swap it if necessary

Reply:
done

Reply:
I don't see it used here in 58